src/ConfigFileParser.cpp

```diff
@@ -1,33 +1,37 @@
 #include "ConfigFileParser.hpp"
+#include "utilities.hpp"
+#include <cstddef>
 
 /**
  * @brief PUBLIC Construct a new ConfigFileParser:: ConfigFileParser object.
  */
-ConfigFileParser::ConfigFileParser(void) { }
-
-/**
- * @brief PRIVATE Construct a new ConfigFileParser:: ConfigFileParser object with copy.
- *
- * The copy constructor is private to prevent direct object creation.
- * @param ref The ConfigFileParser object to copy
- */
-ConfigFileParser::ConfigFileParser(const ConfigFileParser& ref)
+ConfigFileParser::ConfigFileParser(void)
+	: m_configFile()
+	, m_serverIndex(0)
+	, m_locationIndex(0)
 {
-    static_cast<void>(ref);
+	const char* validServerDirectiveNames[]
+		= { "server_name", "listen", "host", "client_max_body_size", "error_page", "location", "root" };
+	const int validServerDirectiveNamesSize = sizeof(validServerDirectiveNames) / sizeof(validServerDirectiveNames[0]);
+
+	const char* validLocationDirectiveNames[] = { "root", "index", "cgi_ext", "cgi_path", "client_max_body_size",
+		"autoindex", "limit_except", "location", "return" };
+	const int validLocationDirectiveNamesSize
+		= sizeof(validLocationDirectiveNames) / sizeof(validLocationDirectiveNames[0]);
+
+	m_validServerDirectives = std::vector<std::string>(validServerDirectiveNames,
+		validServerDirectiveNames
+			+ validServerDirectiveNamesSize); // NOLINT: no good alternative for pointer arithmetic
+	m_validLocationDirectives = std::vector<std::string>(validLocationDirectiveNames,
+		validLocationDirectiveNames
+			+ validLocationDirectiveNamesSize); // NOLINT: no good alternative for pointer arithmetic
 }
 
 /**
- * @brief PRIVATE Overload of the assignment operator.
+ * @brief PUBLIC Destruct a Config File Parser:: Config File Parser object
  *
- * The assigment operator is private to prevent direct object creation.
- * @param ref The ConfigFileParser object to assign.
- * @return ConfigFileParser&  The ConfigFileParser object.
  */
-ConfigFileParser& ConfigFileParser::operator=(const ConfigFileParser& ref)
-{
-    static_cast<void>(ref);
-    return *this;
-}
+ConfigFileParser::~ConfigFileParser(void) { }
```

Comment:
Can be omitted and just have a declaration in header.

Reply:
I don't think that's possible. 

I removed the implementation of the destructor and attempted to compile. 
Then the following error message was displayed:
```
link binary [webserv]
/usr/bin/ld: obj/main.o: in function `main':
main.cpp:(.text+0x12c): undefined reference to `ConfigFileParser::~ConfigFileParser()'
/usr/bin/ld: main.cpp:(.text+0x18c): undefined reference to `ConfigFileParser::~ConfigFileParser()'
collect2: error: ld returned 1 exit status
make: *** [Makefile:152: webserv] Error 1
```
The compiler is complaining because it expects a definition for the destructor, which was declared in the header file.

Reply:
You need to remove the destructor declaration in the header file as well for the compile to use the default destructor. As your destructor does nothing in particular, you can just use the default one. Therefore you can also remove the copy constructor and assignment operator declarations in the header.

Reply:
True. 

I thought that the declaration of the destructor was needed, because there was clang-tidy info, that a copy ctor and assignment operator were defined but not a destructor. 

However, taking in consideration that we now remove redundant declarations of copy ctors and assignment operators, this message does not get displayed anymore. 

Is done.

Reply:
The question is: do you want your ConfigFileParser to be a singleton (only one instance can be created in the course of the whole program)? 

If the answer is yes, then you should have the dtor, copy ctor and assign operator declared as private in order to prevent any of those being called externally. Then you also need the definition of the dtor, but not of the copy ctor and assign operator as it seems (interestingly).

If not, then you can leave everything away if you don't perform any special operations in them.

Reply:
I don't think it is necessary to make the class a singleton. Especially because it seems to be the consent that singletons should be avoided when not needed. 

I will leave the already made changes as they are:

- Removed the declaration and implementation of the destructor
- Removed the declaration of the copy constructor and assignment operator